hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/repository/PrisonerRepository.kt

```diff
@@ -96,6 +140,34 @@ class PrisonerRepository(
     val alias = client.indices().getAlias(GetAliasesRequest().aliases(PRISONER_INDEX), RequestOptions.DEFAULT)
     return alias.aliases.keys
   }
+
+  @Suppress("UNCHECKED_CAST")
+  private fun GetResponse.toPrisonerDocumentSummary(prisonerNumber: String): PrisonerDocumentSummary {
+    return PrisonerDocumentSummary(
+      prisonerNumber,
+      source?.get("bookingId")?.toString()?.toLong(),
+      (source?.get("currentIncentive") as Map<String, *>?)
+        ?.let { currentIncentive ->
+          CurrentIncentive(
+            (currentIncentive["level"] as Map<String, String>)
+              .let { level -> IncentiveLevel(level["code"] as String, level["description"] as String) },
+            LocalDateTime.parse(currentIncentive["dateTime"] as String),
+            LocalDate.parse(currentIncentive["nextReviewDate"] as String),
```

Comment:
This is messy as the low level get api returns the prisoner doc as a map (and nested maps)

Reply:
I have a feeling Jackson has something to convert a Map to an Object - could be wrong though :-)

Reply:
Youre right! Have slotted that in 😄